Source/missiles.cpp

```diff
@@ -3895,109 +3880,112 @@ void MI_Reflect(int i)
+	int dmg = 16 * (GenerateRnd(10) + GenerateRnd(10) + lvl + 2) / 2;
+
+	for (int j = CrawlTable[b]; j > 0; j--, k += 2) {
+		int tx = missile[i]._miVar1 + CrawlTable[k - 1];
+		int ty = missile[i]._miVar2 + CrawlTable[k];
+		if (tx <= 0 || tx >= MAXDUNX || ty <= 0 || ty >= MAXDUNY)
+			continue;
+
+		int dp = dPiece[tx][ty];
+		if (nSolidTable[dp] || dObject[tx][ty] != 0)
+			continue;
+
+		if (!LineClearMissile(missile[i].position.tile.x, missile[i].position.tile.y, tx, ty))
+			continue;
+
+		if (nMissileTable[dp] || missile[i]._miVar8) {
+			missile[i]._miVar8 = 1;
+			continue;
 		}
+
+		AddMissile(tx, ty, tx, ty, 0, MIS_FIREWALL, TARGET_BOTH, src, dmg, missile[i]._mispllvl);
 	}
 }
 
 void MI_LightningRing(int i)
 {
-	int src, tx, ty, dmg, k, j, dp, b;
-	BYTE lvl;
-
-	b = CrawlNum[3];
+	int b = CrawlNum[3];
 	missile[i]._miDelFlag = true;
-	src = missile[i]._micaster;
-	k = CrawlNum[3] + 1;
-	if (src > 0)
-		lvl = plr[src]._pLevel;
-	else
-		lvl = currlevel;
-	dmg = 16 * (GenerateRnd(10) + GenerateRnd(10) + lvl + 2) / 2;
-	for (j = CrawlTable[b]; j > 0; j--, k += 2) {
-		tx = missile[i]._miVar1 + CrawlTable[k - 1];
-		ty = missile[i]._miVar2 + CrawlTable[k];
-		if (tx > 0 && tx < MAXDUNX && ty > 0 && ty < MAXDUNY) {
-			dp = dPiece[tx][ty];
-			if (!nSolidTable[dp] && dObject[tx][ty] == 0) {
-				if (LineClearMissile(missile[i].position.tile.x, missile[i].position.tile.y, tx, ty)) {
-					if (nMissileTable[dp] || missile[i]._miVar8)
-						missile[i]._miVar8 = 1;
-					else
-						AddMissile(tx, ty, tx, ty, 0, MIS_LIGHTWALL, TARGET_BOTH, src, dmg, missile[i]._mispllvl);
-				}
-			}
+	int src = missile[i]._micaster;
+	int k = CrawlNum[3] + 1;
+	uint8_t lvl = (src > 0) ? plr[src]._pLevel : currlevel;
+	int dmg = 16 * (GenerateRnd(10) + GenerateRnd(10) + lvl + 2) / 2;
+	for (int j = CrawlTable[b]; j > 0; j--, k += 2) {
+		int tx = missile[i]._miVar1 + CrawlTable[k - 1];
+		int ty = missile[i]._miVar2 + CrawlTable[k];
+		if (tx <= 0 || tx >= MAXDUNX || ty <= 0 || ty >= MAXDUNY)
+			continue;
+
+		int dp = dPiece[tx][ty];
+		if (nSolidTable[dp] || dObject[tx][ty] != 0)
+			continue;
+
+		if (!LineClearMissile(missile[i].position.tile.x, missile[i].position.tile.y, tx, ty))
+			continue;
+
+		if (nMissileTable[dp] || missile[i]._miVar8) {
+			missile[i]._miVar8 = 1;
+			continue;
 		}
+
+		AddMissile(tx, ty, tx, ty, 0, MIS_LIGHTWALL, TARGET_BOTH, src, dmg, missile[i]._mispllvl);
 	}
 }
 
 void MI_Search(int i)
 {
 	missile[i]._mirange--;
-	if (missile[i]._mirange == 0) {
-		missile[i]._miDelFlag = true;
-		PlaySfxLoc(IS_CAST7, plr[missile[i]._miVar1].position.tile.x, plr[missile[i]._miVar1].position.tile.y);
-		AutoMapShowItems = false;
-	}
+	if (missile[i]._mirange != 0)
+		return;
+
+	missile[i]._miDelFlag = true;
+	PlaySfxLoc(IS_CAST7, plr[missile[i]._miVar1].position.tile.x, plr[missile[i]._miVar1].position.tile.y);
+	AutoMapShowItems = false;
 }
 
 void MI_LightningWallC(int i)
 {
+	int dp, tx, ty;

```

Comment:
Normally you join declaration and assignment. Why did you separate them?

Reply:
To make the left/right growth code as identical as possible and to indicate that it needs some proper refactoring.

Reply:
Have it done, will send it as the next one. Will be a nice small on in comparison :D